packages/cln_plugin/lib/src/cln_plugin_base.dart

```diff
@@ -1,6 +1,47 @@
 // TODO: Put public facing types in this file.
 
-/// Checks if you are awesome. Spoiler: you are.
-class Awesome {
-  bool get isAwesome => true;
+import 'dart:io';
+
+import 'icln_plugin_base.dart';
+
+class Plugin implements ICLNPlugin {
+  // Data members
+  /*
+  ManifestMethod _manifest;
+  List<Interceptor> _preInterceptor;
+  List<Interceptor> _postInterceptor;
+  Map<String, Object> parameters;
+  bool _parametersReady;
+  JsonConverter converter;
+  CLightningPluginConfig configs;
+   */
+
+  Plugin() {
+    // Data member initializations
+  }
+
+  @override
+  void start() {
+    // This function will communicate with the rpc.
+    print("The plugin is started.");
+    try {
+      String? messageSocket;
+      while((messageSocket = stdin.readLineSync()) != null) {
```

Comment:
At some point, we would like to make the plugin async and not sync, this is one limitation of my API in Java!

Sorry!